tests/Microsoft.Bot.Streaming.Tests/StreamingRequestHandlerTests.cs

```diff
@@ -21,16 +21,32 @@ namespace Microsoft.Bot.Builder.Streaming.Tests
 {
     public class StreamingRequestHandlerTests
     {
-        [Fact]
-        public void CanBeConstructedWithANamedPipe()
+        public static IEnumerable<object[]> GetSuccessfulConstructorTestData(int scenario)
+        {
+            var testData = new List<object[]>
+            {
+                new object[] { Guid.NewGuid().ToString(), null },
+                new object[] { Guid.NewGuid().ToString(), "audience" },
+                new object[] { new FauxSock(), null },
+                new object[] { new FauxSock(), "audience" }
+            };
+            
+            return new List<object[]> { testData[scenario] };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetSuccessfulConstructorTestData), parameters: 0)]
+        [MemberData(nameof(GetSuccessfulConstructorTestData), parameters: 1)]
+        public void CanBeConstructedWithANamedPipe(string namedPipe, string audience)
```

Comment:
Instead of writing two new tests for the new constructors that require `string audience`, I used Xunit theories and refactored the tests and private class FauxSock.